assets/src/edit-story/app/media/media3p/useFetchMediaEffect.js

```diff
@@ -84,19 +84,14 @@ export default function useFetchMediaEffect({
       fetchMediaStart({ provider, pageToken });
       try {
         let media, nextPageToken;
-        if (selectedCategoryId) {
-          ({ media, nextPageToken } = await listCategoryMedia({
-            provider,
-            selectedCategoryId,
-            pageToken,
-          }));
-        } else {
-          ({ media, nextPageToken } = await listMedia({
-            provider,
-            searchTerm,
-            pageToken,
-          }));
-        }
+        ({ media, nextPageToken } = await listMedia({
+          provider,
+          filter: {
+            searchTerm: selectedCategoryId ? null : searchTerm,
```

Comment:
constructFilter is already checking that searchTerm and categoryId are not both provided, so perhaps it's easier to not have this check?

Reply:
They're both mutually exclusive in constructFilter, but not in the media3p state which maintains the previous search term.